pallets/xor-fee/src/lib.rs

```diff
@@ -726,7 +813,24 @@ pub mod pallet {
         /// New multiplier for weight to fee conversion is set
         /// (*1_000_000_000_000_000_000). [New value]
         WeightToFeeMultiplierUpdated(FixedU128),
+        #[cfg(feature = "ready-to-test")] // Dynamic fee
+        /// New block number to update multiplier is set. [New value]
+        NextUpdateBlockUpdated(Option<<T as frame_system::Config>::BlockNumber>),
     }
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Failed to calculate new multiplier.
+        MultiplierCalculationFailed,
+    }
+
+    #[cfg(feature = "ready-to-test")] // Dynamic fee
+    /// Next block number to update multiplier
+    /// If it is necessary to stop updating the multiplier,
+    /// set BlockNumber::MAX or another large value
+    #[pallet::storage]
+    #[pallet::getter(fn next_update_block)]
+    pub type NextUpdateBlock<T> =
```

Comment:
It is better to use `LastUpdateBlock` and check `current_block >= last_update_block + update_period`. In this case if you change parameter `update_period`, you don't need to change the `NextUpdateBlock`.

Reply:
Decided to use only period